main.c

```diff
@@ -76,116 +118,185 @@ int main(void) {
+#if (BOARD_UNIQUE_ID == BOARD_ID_CHARGING_AIRBRAKE || BOARD_UNIQUE_ID == BOARD_ID_CHARGING_PAYLOAD)
+            update_motor_curr_low_pass();
+#elif (BOARD_UNIQUE_ID == BOARD_ID_CHARGING_CAN)
+            update_5v_curr_low_pass();
+            update_13v_curr_low_pass();
+#endif            
         }
+
+        #if (BOARD_UNIQUE_ID == BOARD_ID_CHARGING_AIRBRAKE)
+            // state transition from boost to coast, enable motor
+            if (state == BOOST && ((millis() - inj_open_time) > BOOST_LENGTH_MS)) {
+                state = COAST;
+                MOTOR_POWER = MOTOR_ON;
+            }
+        
+        
+            //state transition from coast to descent, enable motor for MOTOR_ACT_TIME_MS
+            if (state == COAST && ((millis() - inj_open_time) > (BOOST_LENGTH_MS + COAST_LENGTH_MS))) {
+                state = DESCENT;
+            
+                MOTOR_POWER = MOTOR_ON;
+                cmd_airbrakes_ext = 0;
+                airbrakes_act_time = millis();
+            }
+        
+            //If we are on the ground or in descent, cut motor power after a certain period of time
+            if ((state == PRE_FLIGHT || state == DESCENT) 
+                && ((millis() - airbrakes_act_time) > MOTOR_ACT_TIME_MS)) {
+            
+                MOTOR_POWER = !MOTOR_ON;
+                cmd_airbrakes_ext = 0;
+            }
+
+        #endif
     }
 }
 
 static void can_msg_handler(const can_msg_t *msg) {
     seen_can_message = true;
     uint16_t msg_type = get_message_type(msg);
+    
+    //echo -n "m0C0,00,00,00,01,00;" > /dev/tty.usbmodem12401                   //injector valve open message
+    //echo -n "m220,00,00,00,09,00,00,80,3F;" > /dev/tty.usbmodem21401          //actuate to 1
+    //echo -n "m220,00,00,00,09,66,66,66,3F;" > /dev/tty.usbmodem21401          //actuate to 0.9
+    //echo -n "m220,00,00,00,09,00,00,00,3F;" > /dev/tty.usbmodem21401          //actuate to 0.5
+    //echo -n "m220,00,00,00,09,DC,CC,CC,3E;" > /dev/tty.usbmodem21401          //actuate to 0.4
+    //echo -n "m220,00,00,00,09,00,00,00,00;" > /dev/tty.usbmodem21401          //actuate to 0.0
 
     // ignore messages that were sent from this board
-    if (get_board_unique_id(msg) == BOARD_UNIQUE_ID) {
+    if (get_board_unique_id(msg) == BOARD_UNIQUE_ID || get_board_unique_id(msg)==BOARD_ID_LOGGER) {
         return;
     }
 
     int act_id;
     int act_state;
     int dest_id;
+
     switch (msg_type) {
-        case MSG_ACTUATOR_CMD:
+        case MSG_ACTUATOR_CMD: // this will toggle *all* battery chargers, not just CHARGING_CAN
             act_id = get_actuator_id(msg);
             act_state = get_req_actuator_state(msg);
-            if (act_id == ACTUATOR_CANBUS) {
+            
+            //Battery Charger On/Off
+            if (act_id == ACTUATOR_CHARGE) {
+                if (act_state == ACTUATOR_ON) {
+                    BATTERY_CHARGER_EN(true);
+                    BLUE_LED_SET(true); //temporarily commented out
```

Comment:
what does the comment mean? is this meant to be here or not?

Reply:
I'm assuming no since it has overlap with the motor power implementation in the main loop?